packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -6153,6 +6126,83 @@ addEventListener('fetch', event => {});`
 				expect(std.err).toMatchInlineSnapshot(`""`);
 				expect(std.warn).toMatchInlineSnapshot(`""`);
 			});
+
+			it("should error for compiled logfwdr schemas", async () => {
+				writeWranglerToml({
+					logfwdr: {
+						schema: "./message.capnp.compiled",
+						bindings: [
+							{
+								name: "httplogs",
+								destination: "httplogs",
+							},
+							{
+								name: "trace",
+								destination: "trace",
+							},
+						],
+					},
+				});
+
+				await expect(() => runWrangler("deploy index.js")).rejects
+					.toThrowErrorMatchingInlineSnapshot(`
+			"Processing wrangler.toml configuration:
+			  - \\"logfwdr\\" bindings \\"schema\\" field should not be pre-compiled - Wrangler will run capnp for you."
+		`);
+			});
+
+			it("should when a logfwdr schema isn't specified", async () => {
+				writeWranglerToml({
+					logfwdr: {
+						schema: undefined,
+						bindings: [
+							{
+								name: "httplogs",
+								destination: "httplogs",
+							},
+							{
+								name: "trace",
+								destination: "trace",
+							},
+						],
+					},
+				});
+
+				await expect(() => runWrangler("deploy index.js")).rejects
+					.toThrowErrorMatchingInlineSnapshot(`
+			"Processing wrangler.toml configuration:
+			  - \\"logfwdr\\" bindings should have a string \\"schema\\" field but got {\\"bindings\\":[{\\"name\\":\\"httplogs\\",\\"destination\\":\\"httplogs\\"},{\\"name\\":\\"trace\\",\\"destination\\":\\"trace\\"}]}."
+		`);
+			});
+
+			it("should error when the capnp CLI tool is not present", async () => {
```

Comment:
I might be able to also add a test for when it is present, which then mocks it's output - however this seems rather tricky, so I might need a little help